CMakeLists.txt

```diff
@@ -47,13 +47,6 @@ SET(CMAKE_ALLOW_LOOSE_LOOP_CONSTRUCTS true)
 SET(CMAKE_CXX_STANDARD 11)
 SET(CMAKE_CXX_STANDARD_REQUIRED ON)
 
-if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
```

Comment:
you could leave that in, but change the version to `>=4.7` (which is what we require now, if I remember correctly)

We should also think of a Tag, such as `GCC-OPT 4.X` which we mention right here for documentation and whenever we increase this version, we can go and check for this tag in our codebase.
You could, for example, tag the current workaround for var.templates and this would remind us to revert back to the nicer var.templ. code in the future.